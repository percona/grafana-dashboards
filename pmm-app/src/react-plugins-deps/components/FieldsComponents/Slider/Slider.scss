#antd {
  .ant-slider-step {
    display: none;
  }

  .ant-slider-rail {
    background-color: rgb(40, 40, 40);
    border-radius: 5px;
    height: 10px;
  }

  .ant-slider-handle {
    height: 20px;
    width: 20px;
  }

  .ant-slider-mark {
    margin-top: 5px;
  }
}
