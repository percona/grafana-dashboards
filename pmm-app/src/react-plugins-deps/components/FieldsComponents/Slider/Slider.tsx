import React from 'react';
import { useField } from 'react-final-form-hooks';
import { Slider } from 'antd';
import './Slider.scss';

const { Option } = Slider;
interface SliderFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form?: any;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
  marks?: any;
}
export const SliderField = ({ name, marks: marks, form, style, defaultValue }: SliderFieldInterface) => {
  const { input } = useField(name, form);
  return (
    <span className={'fields__Slider-field'} style={style || {}}>
      <Slider {...input} marks={marks} max={2} step={null} included={false} defaultValue={defaultValue} />
    </span>
  );
};
