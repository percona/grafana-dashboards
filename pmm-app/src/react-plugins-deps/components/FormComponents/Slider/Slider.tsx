import React from 'react';
import { useField } from 'react-final-form';
import { Slider } from 'antd';
import './Slider.scss';

interface SliderFieldInterface {
  name: string;
  style?: any;
  defaultValue?: any;
  marks?: any;
  tipFormatter: any;
}
export const SliderField = ({
  name,
  marks: marks,
  style,
  defaultValue,
  tipFormatter,
}: SliderFieldInterface) => {
  const { input } = useField(name);
  return (
    <span className="fields__Slider-field" style={style || {}}>
      <Slider
        {...input}
        marks={marks}
        max={2}
        step={null}
        included={false}
        defaultValue={defaultValue}
        tipFormatter={tipFormatter}
      />
    </span>
  );
};
