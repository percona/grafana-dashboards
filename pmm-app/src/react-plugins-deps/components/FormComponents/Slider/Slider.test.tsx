// Just a stub test
import { SliderField } from './Slider';
import React from 'react';
import renderer from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks';
import { Form } from 'react-final-form';

describe('Slider field test', () => {
  it('Slider renders correct without props', () => {
    renderHook(() => {
      const marks = {
        0: 'Low',
        1: 'Medium',
        2: 'High',
      };

      const component = renderer.create(
        <Form
          onSubmit={() => null}
          render={() => <SliderField marks={marks} defaultValue={2} name="metrics_resolutions_slider" />}
        />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
