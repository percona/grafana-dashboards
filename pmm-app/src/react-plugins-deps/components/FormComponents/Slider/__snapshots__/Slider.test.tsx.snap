// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider field test Slider renders correct without props 1`] = `null`;
